azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultOperation.java

```diff
@@ -111,12 +119,17 @@ private String toKeyVaultSecretName(@NonNull String property) {
 
     public String get(final String property) {
         Assert.hasText(property, "property should contain text.");
-        refreshSecretKeysIfNeeded();
-        return Optional.of(property)
-                .map(this::toKeyVaultSecretName)
-                .filter(secretNames::contains)
-                .map(this::getValueFromKeyVault)
-                .orElse(null);
+        final String secretName = toKeyVaultSecretName(property);
+        if (!refreshAndGetSecretNames().contains(secretName)) {
+            return null;
+        }
+        final String cachedValue = keyVaultVaultCache.getOrDefault(secretName, null);
+        if (cachedValue != null) {
+            return cachedValue;
+        }
```

Comment:
Not use following code to avoid multi-thread problem.
```
if (keyVaultVaultCache.contains(secretName)){
  return keyVaultVaultCache.get(secretName);
}
```